### PR TITLE
Scastie Embedding blog post and Front Page Embedding

### PR DESCRIPTION
**Pull request by MasseGuillaume**
Superseed #749

### CODE REVIEW

**Comment by MasseGuillaume** [comment]
out

File: _includes/scastie.html [hunk 1]
```diff
@@ -14,16 +14,32 @@
 	<div class="call-to-action action-medium">
 		<a href="{{site.data.common.scastieUrl}}" class="button" target="_blank">Learn More</a>
 	</div>
+
 	<div class="wrap">
 		<div class="code-element">
 			<div class="bar-code"></div>
 			<div class="scastie-code-container">
-				<textarea id="scastie-textarea"></textarea>
+        <script src="https://scastie.scala-lang.org/wG96cURqRlmLoOFFa8vHtA.js?theme=dark"></script>
```
Comment (julienrf) [inline _includes/scastie.html]: Really? Why didn’t you pick one of the snippets submitted at the contest?

Comment (MasseGuillaume) [reply _includes/scastie.html]: A ops, will do! Which one you think should be it?

File: _posts/2017-09-25-scastie-embedding.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+This is all possible because Scastie is based on sbt, allowing us to support newer
+Scala versions and resolve library dependencies on Maven.
+
+Want to see what Scastie can do? Check out this embedding below!
+
+<script src="https://scastie.scala-lang.org/d4Zr5VLvQAirvPwIr2bqMA.js"></script>
+
+## How do I embed Scastie in my website/documentation
+
+Since it's possible to use external libraries, Scastie is an excellant tool to add to your library documentation or your blog post. Let's explore the available methods to bootstrap the embedding mode.
```
Comment (julienrf) [inline _posts/2017-09-25-scastie-embedding.md]: excell**e**nt

File: _posts/2017-09-25-scastie-embedding.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+## How do I embed Scastie?
```
Comment (julienrf) [inline _posts/2017-09-25-scastie-embedding.md]: Do we really need this heading? If yes, why is it at the same level as the previous one?

Comment (MasseGuillaume) [reply _posts/2017-09-25-scastie-embedding.md]: ops

File: _posts/2017-09-25-scastie-embedding.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+<!-- Annonymous Resource -->
+<script src="https://scastie.scala-lang.org/xbrvky6fTjysG32zK6kzRQ.js"></script>
+
+<!-- User's Resource -->
+<script src="https://scastie.scala-lang.org/MasseGuillaume/CpO2s8v2Q1qGdO3vROYjfg.js"></script>
+<script src="https://scastie.scala-lang.org/MasseGuillaume/CpO2s8v2Q1qGdO3vROYjfg/1.js"></script>
+```
+
+The embedding will be patched at the location of the script tag.
```
Comment (julienrf) [inline _posts/2017-09-25-scastie-embedding.md]: What does “patching an embedding” mean?

File: _posts/2017-09-25-scastie-embedding.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+It's also possible to adjust the theme via the GET parameter theme. For example:
```
Comment (julienrf) [inline _posts/2017-09-25-scastie-embedding.md]: missing backticks around `theme`

File: _posts/2017-09-25-scastie-embedding.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+```html
+<script src="https://scastie.scala-lang.org/xbrvky6fTjysG32zK6kzRQ.js?theme=dark"></script>
+```
+
+The theme can be `dark` or `light`. The default value is `light`.
+
+Finally, it's also possible to customize the css such that it blends into your website.
```
Comment (julienrf) [inline _posts/2017-09-25-scastie-embedding.md]: **CSS** (caps)

File: _posts/2017-09-25-scastie-embedding.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+Take a look at: [https://codepen.io/MasseGuillaume/pen/oeKoee](https://codepen.io/MasseGuillaume/pen/oeKoee)
+
+#### Via Javascript
```
Comment (julienrf) [inline _posts/2017-09-25-scastie-embedding.md]: Java**S**cript

File: _posts/2017-09-25-scastie-embedding.md (hunk 4 above)
Comment (MasseGuillaume) [reply _posts/2017-09-25-scastie-embedding.md]: I will use the word inserted.

File: _posts/2017-09-25-scastie-embedding.md (hunk 5 above)
Comment (MasseGuillaume) [reply _posts/2017-09-25-scastie-embedding.md]: done

File: _posts/2017-09-25-scastie-embedding.md (hunk 6 above)
Comment (MasseGuillaume) [reply _posts/2017-09-25-scastie-embedding.md]: done

File: _posts/2017-09-25-scastie-embedding.md (hunk 7 above)
Comment (MasseGuillaume) [reply _posts/2017-09-25-scastie-embedding.md]: Done.

File: _posts/2017-09-25-scastie-embedding.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+```html
+
+<!-- Embedding Script -->
+<div id="resource"></div>
+<script src="https://scastie.scala-lang.org/embedded.js"></script>
+
+<script>
+window.addEventListener('load', function() {
+  scastie.Embedded('#resource-1', {
```
Comment (julienrf) [inline _posts/2017-09-25-scastie-embedding.md]: Is it expected that the selector doesn’t match the id of the div above?

File: _posts/2017-09-25-scastie-embedding.md (hunk 2 above)
Comment (MasseGuillaume) [reply _posts/2017-09-25-scastie-embedding.md]: Thank you :)

File: _posts/2017-09-25-scastie-embedding.md (hunk 8 above)
Comment (MasseGuillaume) [reply _posts/2017-09-25-scastie-embedding.md]: ops, fixed

File: _includes/scastie.html (hunk 1 above)
Comment (MasseGuillaume) [reply _includes/scastie.html]: I like this one: https://scastie.scala-lang.org/dOrqJ7o7Q06dm37Zmb7vKg

**Comment by MasseGuillaume** [comment]
@SethTisue or @heathermiller I will need your blessing on this PR!

**Comment by heathermiller** [comment]
Hey @MasseGuillaume I'd like to be the one to merge this; I've also got to do @olafurpg's as well. I'm speaking at Strange Loop today, and am a bit overwhelmed with that. I can do this right afterwards, but by then it will probably be too late.

It might be better to hold off on this and @olafurpg's until Monday.

Sorry guys, today and yesterday have been a bit nuts for me.

**Comment by MasseGuillaume** [comment]
Don't worry, take it easy.

**Comment by olafurpg** [comment]
No worries @heathermiller enjoy Strange Loop!

File: _includes/scastie.html (hunk 1 above)
Comment (nafg) [reply _includes/scastie.html]: I get `//snippet not found`

**Comment by MasseGuillaume** [comment]
@heathermiller ready for review.

**Pull request closed without merging**